custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml

```diff
@@ -7,18 +8,114 @@ primary_entity:
     - id: 1
       name: switch
       type: boolean
+      mapping:
+        - dps_val: true
+          icon: mdi:scent
+          value: ON
+        - dps_val: false
+          icon: mdi:scent-off
+          value: OFF      
     - id: 2
       name: speed
       type: string
+      hidden: true
       mapping:
         - dps_val: large
           value: 100
+          icon: "mdi:sprinkler-variant"
         - dps_val: small
           value: 50
+          icon: "mdi:sprinkler"
+    - id: 4          
+      name: "Left time"
+      readolny: true
+      optional: true
+      type: integer
+      range:
+        min: 0
+        max: 360
 secondary_entities:
   - entity: light
     name: Nightlight
+    icon: "mdi:light-recessed"
     dps:
       - id: 5
         name: switch
         type: boolean
+      - id: 6 
+        name: color_mode  #effect
+        type: string
+        hidden: true
+        mapping:
+          - dps_val: colour
+            value: rgbw 
+      - id: 8   # not working
+        name: rgbhsv  
+        type: hex     
+        format:
+          - name: r
+            bytes: 1
+          - name: g
+            bytes: 1
+          - name: b
+            bytes: 1
+          - name: h
+            bytes: 2
+            range:
+              min: 0
+              max: 360
+          - name: s
+            bytes: 1
+            range:
+              min: 0
+              max: 100
```

Comment:
It appears from the comment at the top of the file to have a range of 0 to 255.  The RGB and HSV values appear to be a close match, so I think the format is correct apart from that, and this would be the same as the atomi_string_lights.yaml format.

Reply:
Fixed (this was copied from other integration). Anyway even with this selecting color doesn't work because device only supports hs, and not rgbw mode.